x/epoching/keeper/epoch_msg_queue.go

```diff
@@ -341,3 +341,70 @@ func cacheTxContext(ctx sdk.Context, txid []byte, msgid []byte, height uint64) (
 
 	return ctx.WithMultiStore(msCache), msCache
 }
+
+// ClearEpochMsgs clears all processed messages and queue metadata for a given epoch
+func (k Keeper) ClearEpochMsgs(ctx context.Context, epochNumber uint64) error {
+	// Get queue length first for logging
+	if epochNumber == 0 {
+		return errorsmod.Wrap(types.ErrInvalidEpoch, "epoch number cannot be zero")
+	}
+
+	currentEpoch := k.GetEpoch(ctx).EpochNumber
+	if epochNumber >= currentEpoch {
+		return errorsmod.Wrapf(types.ErrInvalidEpoch,
+			"cannot clear current or future epoch %d (current: %d)", epochNumber, currentEpoch)
+	}
+
+	queueLength := k.GetQueueLength(ctx, epochNumber)
+
+	if queueLength == 0 {
+		// No messages to clear
+		return nil
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	// Clear all messages in the queue store
+	store := k.msgQueueStore(ctx, epochNumber)
+
+	// Iterate and delete all messages
+	iterator := storetypes.KVStorePrefixIterator(store, nil)
+	defer func() {
+		if err := iterator.Close(); err != nil {
+			sdkCtx.Logger().Error("Failed to close iterator", "error", err)
+		}
+	}()
+
+	keysToDelete := [][]byte{}
+	for ; iterator.Valid(); iterator.Next() {
+		keysToDelete = append(keysToDelete, iterator.Key())
+	}
+
+	// Delete all message entries
+	for _, key := range keysToDelete {
+		store.Delete(key)
+	}
+
+	// Clear the queue length entry
+	lengthStore := k.msgQueueLengthStore(ctx)
+	epochNumberBytes := sdk.Uint64ToBigEndian(epochNumber)
+	lengthStore.Delete(epochNumberBytes)
+
+	// Verify the cleanup was successful by checking queue length
+	remainingLength := k.GetQueueLength(ctx, epochNumber)
+	if remainingLength != 0 {
+		return errorsmod.Wrapf(types.ErrCleaningQueue,
```

Comment:
If deletion doesn’t cause serious issues when it fails, and it’s evident that deletion should succeed, it might be enough to just verify it in the test code instead of handling the error explicitly.

Since it’s a function called in the EndBlocker, if it throws an error, it will halt the chain.